### ISSUE TITLE
System.NullReferenceException in NestedMarkupExtension.ProvideValue

### ISSUE THREAD

**Issue by Dunge** [XAMLMarkupExtensions/XAMLMarkupExtensions#101]
I'm porting a .NET 4.5 WPF project to .NET5

I updated the WPFLocalizeExtension nuget package from 1.0.0 to 3.8.0 and then changed all xlmns references from ""clr-namespace:WPFLocalizeExtension.Extensions;assembly=WPFLocalizeExtension" to "http://wpflocalizeextension.codeplex.com".

It seemed to work for a small outside project as a test, but when starting my main application I get the following exception:

```
   at XAMLMarkupExtensions.Base.NestedMarkupExtension.ProvideValue(IServiceProvider serviceProvider)
   at MS.Internal.Xaml.Runtime.ClrObjectRuntime.CallProvideValue(MarkupExtension me, IServiceProvider serviceProvider)
   at System.Xaml.XamlObjectWriter.Logic_ProvideValue(ObjectWriterContext ctx)
   at System.Xaml.XamlObjectWriter.Logic_AssignProvidedValue(ObjectWriterContext ctx)
   at System.Xaml.XamlObjectWriter.WriteEndObject()
   at System.Xaml.XamlWriter.WriteNode(XamlReader reader)
   at System.Windows.Markup.WpfXamlLoader.TransformNodes(XamlReader xamlReader, XamlObjectWriter xamlWriter, Boolean onlyLoadOneNode, Boolean skipJournaledProperties, Boolean shouldPassLineNumberInfo, IXamlLineInfo xamlLineInfo, IXamlLineInfoConsumer xamlLineInfoConsumer, XamlContextStack`1 stack, IStyleConnector styleConnector)
   at System.Windows.Markup.WpfXamlLoader.Load(XamlReader xamlReader, IXamlObjectWriterFactory writerFactory, Boolean skipJournaledProperties, Object rootObject, XamlObjectWriterSettings settings, Uri baseUri)
   at System.Windows.Markup.WpfXamlLoader.LoadBaml(XamlReader xamlReader, Boolean skipJournaledProperties, Object rootObject, XamlAccessLevel accessLevel, Uri baseUri)
   at System.Windows.Markup.XamlReader.LoadBaml(Stream stream, ParserContext parserContext, Object parent, Boolean closeStream)
   at System.Windows.Application.LoadComponent(Object component, Uri resourceLocator)
   at Views.UserControls.FilteredListUserControl.InitializeComponent() in FilteredListUserControl.xaml:line 1
```

Is this because one referenced localization string doesn't exists in my LocalizationRessource.resx file?

Could the exception message be a bit more explicit and actually show which string is missing?

**Comment by Dunge** [comment]
I tracked down the error to using LocTex in a DataTrigger setter. The localization string does exist. This is something that was working properly in WPFLocalizeExtension 1.0.0 :

     <DataTemplate>
             <TextBlock x:Name="LastActivationText"
                        Text="{Binding LastActivation, Converter={infrastructure:InjectedConverter DateTimeToStringTimeZoneConverter}}" />
         <DataTemplate.Triggers>
             <DataTrigger Binding="{Binding LastActivation}"
                          Value="{x:Static system:DateTime.MaxValue}">
                 <Setter TargetName="LastActivationText"
                         Property="Text"
                         Value="{lex:LocText Client.WpfInterface:LocalizationResource:InProgress}" />
             </DataTrigger>
         </DataTemplate.Triggers>
     </DataTemplate>

**Comment by Karnah** [comment]
Hello, @Dunge.
`Setter` and `Binding` support were added only in 3.8.0 (but probably it were many versions ago) and now has some restrictions. Probably the most serious of them - it doesn't know anything about `ILocalizationProvider` of parent control and it always searches key in `LocalizeDictionary.Instance.DefaultProvider`. If you don't set `DefaultProvider`, please set and check if it will work.

**Comment by Dunge** [comment]
Hello Karnah.

I'm not sure what you are mentioning about looking the provider of parent control, but that's not my goal, in my situation I have only one localizationresource and it is used in the whole application. I don't need to switch.

I managed to fix my issue by replacing 

    Value="{lex:LocText Client.WpfInterface:LocalizationResource:InProgress}"

by

    Value="{Binding Source={lex:LocText VerMac.JamLogic.Client.WpfInterface:LocalizationResource:InProgress}}"

as I found in an example looking at other issues on this github page. I don't know the implications of adding Source there, but it works.

I just wish there would be a better error handling message than triggering an obscure NullReferenceException and not telling you any indication on where the source of the problem lies in your code.
